core/src/test/java/org/apache/brooklyn/core/location/LocationExtensionsTest.java

```diff
@@ -164,21 +166,24 @@ public void testAddExtensionThroughLocationSpecIllegally() {
         try {
             Location loc = createConcrete(MyExtension.class, "not an extension");
             fail("loc="+loc);
-        } catch (IllegalArgumentException e) {
+        } catch (Exception e) {
+            assertTrue(Exceptions.getFirstInteresting(e) instanceof IllegalArgumentException);
```

Comment:
I prefer that `e` is rethrown if it doesn't match, rather than losing the exception in the error report. Alex (I think it was) added:

    Asserts.expectedFailureOfType(e, IllegalArgumentException.class)

But not sure if that deals with looking at the `causedBy` properly.